src/ecdh.c

```diff
@@ -19,13 +19,12 @@ int wally_ecdh(const unsigned char *pub_key, size_t pub_key_len,
         !bytes_out || len != SHA256_LEN)
         return WALLY_EINVAL;
 
-    if (!pubkey_parse(&pub, pub_key, pub_key_len) ||
-        !seckey_verify(priv_key)) {
+    wally_clear(bytes_out, len);
+    if (!pubkey_parse(&pub, pub_key, pub_key_len))
         ret = WALLY_EINVAL;
-    } else if (!secp256k1_ecdh(ctx, bytes_out, &pub, priv_key, NULL, NULL)) {
-        wally_clear(bytes_out, len);
+    else if (!seckey_verify(priv_key) ||
+             !secp256k1_ecdh(ctx, bytes_out, &pub, priv_key, NULL, NULL))
```

Comment:
why don't we clear `bytes_out` if `secp256k1_ecdh` failed?

Reply:
This call can't fail with a validated pub/privkey which we now check before calling it.